submodules/bastion/variables.tf

```diff
@@ -3,56 +3,61 @@ variable "deploy_id" {
   description = "Domino Deployment ID"
 }
 
-variable "ami_id" {
-  description = "AMI ID for the bastion EC2 instance, otherwise we will use the latest 'amazon_linux_2' ami."
-  type        = string
-  default     = null
-}
-
-variable "instance_type" {
-  description = "the bastion's instance type, if null, t2.micro is used"
-  type        = string
-  default     = null
-}
-
 variable "region" {
   description = "AWS region for the deployment"
   type        = string
 }
 
-variable "vpc_id" {
-  description = "VPC ID."
-  type        = string
-}
-
-variable "ssh_pvt_key_path" {
-  description = "SSH private key filepath."
-  type        = string
-}
-
-variable "ssh_key_pair_name" {
-  description = "AWS key_pair name."
-  type        = string
-}
-
-variable "public_subnet_id" {
-  description = "Public subnet to create bastion host in."
-  type        = string
-}
-
-variable "security_group_rules" {
-  description = "Bastion host security group rules."
-  type = map(object({
-    protocol                 = string
-    from_port                = string
-    to_port                  = string
-    type                     = string
-    description              = string
-    cidr_blocks              = list(string)
-    source_security_group_id = string
-  }))
-
-  default = {}
+variable "network_info" {
+  description = <<EOF
+    id = VPC ID.
+    subnets = {
+      public = List of public Subnets.
+      [{
+        name = Subnet name.
+        subnet_id = Subnet ud
+        az = Subnet availability_zone
+        az_id = Subnet availability_zone_id
+      }]
+      private = List of private Subnets.
+      [{
+        name = Subnet name.
+        subnet_id = Subnet ud
+        az = Subnet availability_zone
+        az_id = Subnet availability_zone_id
+      }]
+      pod = List of pod Subnets.
+      [{
+        name = Subnet name.
+        subnet_id = Subnet ud
+        az = Subnet availability_zone
```

Comment:
Part of it is, rather than having data sources everywhere, have the network figure it out and pass it, consumers will use what they need from the `object`